distributed/protocol/core.py

```diff
@@ -122,7 +122,23 @@ def loads(frames, deserialize=True, deserializers=None):
             else:
                 value = Serialized(head, fs)
 
-            get_in(key[:-1], msg)[key[-1]] = value
+            from toolz import get_in
+
+            def put_in(keys, coll, val):
```

Comment:
Alternatively instead of this approach we could go for something like

```python
        msg = loads_msgpack(small_header, small_payload)
        if not frames:
             return msg
```
becomes
```python
if not frames:
    msg = loads_msgpack(small_header, small_payload)
    return msg
msg = loads_msgpack(small_header, small_payload, use_list=True)
```
and then the mutability assumptions further on should hold

Reply:
Ah, now I understand the need for this.  I missed the `get_in` code that this was replacing earlier.  Sorry, I was a bit dense about the need for this. 

I would be suspicious of sometimes using `use_list=True` and sometimes not.  If the `put_in` solution here works and isn't too slow then we should probably stick with it.  We might also consider returning a tuple rather than a list, again if it doesn't cause a performance hit (my guess is that it wouldn't)